### DISCUSSION TITLE
How to use "icontains" in queries

### DISCUSSION THREAD

**Discussion by przwr** [piccolo-orm/piccolo#1035] [Q&A] [answered]
Hi!

I'm continue converting to Piccolo ORM. I can't find way to use SQL "contains" closure. How can I do it? I want to filter by part of a string. I want to do something like:

```
Table.name.icontains("name") 
```

I guess I can use:
```
Table.name.ilike('%name%')
```

but maybe there is a different way of achieving contains?

**Comment by dantownsend** [comment, marked as answer]
Are you using SQLite or Postgres? `like` and `ilike` are probably your best option. I'm not sure if Postgres has `contains`. If it does, then you could use `WhereRaw` to access it.

**Reply by przwr** [reply]
For testing I'm using sqlite but postgres in production. In tortoise orm I was using is_contains and it was working both - in sqlite and postgres, but maybe it did something different in actual sql

**Reply by dantownsend** [reply]
Yeah, I know SQLServer has `contains`, which is a slightly more powerful version of `like`. Tortoise is probably just using `like` under the hood for Postgres and SQLite.
